src/main/java/io/vertx/rabbitmq/RabbitMQClient.java

```diff
@@ -65,26 +65,49 @@ static RabbitMQClient create(Vertx vertx, JsonObject config) {
   void basicGet(String queue, boolean autoAck, Handler<AsyncResult<JsonObject>> resultHandler);
 
   /**
+   * Use {@link RabbitMQClient#basicConsumer(String, Handler)} instead
+   * <p>
    * Start a non-nolocal, non-exclusive consumer, with auto acknowledgement and a server-generated consumerTag.
    *
    * @see com.rabbitmq.client.Channel#basicConsume(String, Consumer)
    */
+  @Deprecated
   void basicConsume(String queue, String address, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Use {@link RabbitMQClient#basicConsumer(String, boolean, Handler)} instead
+   * <p>
    * Start a non-nolocal, non-exclusive consumer, with a server-generated consumerTag.
    *
    * @see com.rabbitmq.client.Channel#basicConsume(String, boolean, String, Consumer)
    */
+  @Deprecated
   void basicConsume(String queue, String address, boolean autoAck, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Use {@link RabbitMQClient#basicConsumer(String, boolean, Handler)} instead
+   * <p>
    * Start a non-nolocal, non-exclusive consumer, with a server-generated consumerTag and error handler
    *
    * @see com.rabbitmq.client.Channel#basicConsume(String, boolean, String, Consumer)
    */
+  @Deprecated
   void basicConsume(String queue, String address, boolean autoAck, Handler<AsyncResult<Void>> resultHandler, Handler<Throwable> errorHandler);
 
+  /**
+   * @param resultHandler handler with result of operation and consumer tag
+   * @return a stream for message consumption
+   * @see com.rabbitmq.client.Channel#basicConsume(String, Consumer)
+   */
+  RabbitMQueue basicConsumer(String queue, Handler<AsyncResult<String>> resultHandler);
```

Comment:
what does the resultHandler provides ?

Reply:
a _String_ is a consumer tag.

Reply:
So if result of operation is success, then a client can retrieve a consumer tag

Reply:
If operation is failed, then a queue won't work and a client should figure out the problem

Reply:
I think instead we should have rather:

```
void basicConsumer(String queue, Handler<AsyncResult<RabbitMQueue>>) 
```

Reply:
yeah, this is making much more sense. Just done